integreat_compass/cms/views/interactions/vote_actions.py

```diff
@@ -0,0 +1,35 @@
+from django.shortcuts import redirect
+from django.views.decorators.http import require_POST
+
+from ...decorators import permission_required
+from ...models import Vote
+
+
+@require_POST
+@permission_required("vote_on_offer")
+def add_vote(request, offer_version_id):
+    """
+    Add vote of a user to an offer version
+
+    :param request: Object representing the user call
+    :type request: ~django.http.HttpRequest
+
+    :param offer_version_id: internal id of the offer_version to be voted on
+    :type offer_version_id: int
+
+    :return: The rendered template response
+    :rtype: ~django.template.response.TemplateResponse
+    """
+    approval = request.POST.get("approval")
+    vote = Vote.objects.filter(creator=request.user).filter(
+        offer_version=offer_version_id
+    )
+    if vote:
+        vote[0].approval = approval
+        vote[0].save()
+    else:
+        new_vote = Vote(
+            creator=request.user, offer_version_id=offer_version_id, approval=approval
+        )
+        new_vote.save()
+    return redirect("/dashboard/votes")
```

Comment:
Could you please also add a success message for the user? Right now it's really hard to tell if the vote was successfully cast. If your vote crosses the threshold, then the application just seems to disappear.

Ideally, I think different messages should be shown depending on:
- was the vote cast for the first time?
- was the vote changed?
- did the vote lead to acceptance/rejection of the offer application?

Reply:
I added a simple success message, if you think more is needed, i think this should be done in a separate ticket and be discussed what is needed / favored.

Reply:
No, I think that's perfectly fine! Just something of a feedback for the user.